### DISCUSSION TITLE
Better solution for permission

### DISCUSSION THREAD

**Discussion by Megafry** [solspace/craft-freeform#602] [Feature Requests]
We have the following scenario, user group A and B.
Both user groups can edit selected forms and create new ones. 
 
- The user from user group B creates a new form and gets permission automatically, but users from the same group do not.
- When a user in groups A and B creates a new form, he should be able to define the permission (if A, B or both).
- When an administrator creates a new form, he can assign it via the user groups, but this changes the config file (git).
- If I create a new user for user group B afterward, he should also get all permissions.
- maybe the permission logic should be stored in the Database and not in the config

What is the best solution for all this issues?

**Comment by kjmartens** [comment]
Thanks for your feedback on this @Megafry.

The dilemma we have is that Project Config is all or nothing, and while we definitely could rework permissions to be in their own separate Freeform database table, that solution would have many other customers upset about it not being a part of Project Config.

Likely, the only solution here would be to have both, along with a setting where the user can choose whether Freeform respects Project Config or Freeform's own hypothetical permission controls.

**Comment by Megafry** [comment]
Thanks for your feedback @kjmartens 

I understand the difficulty 

Another idea I had was to add form folders. 
Then I can create 2 folders, one for user group A and B.

The user B has then only access to folder B and all newly created forms are automatically visible for all users in this group. 

The permissions can then be set either on the forms or folders.
 
Best regards

**Reply by kjmartens** [reply]
That's an interesting idea, @Megafry. I've noted it for future consideration. 🙂
